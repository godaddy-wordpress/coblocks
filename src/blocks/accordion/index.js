/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
