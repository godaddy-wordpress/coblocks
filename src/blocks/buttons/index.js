/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
