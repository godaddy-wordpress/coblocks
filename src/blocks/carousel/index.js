/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
