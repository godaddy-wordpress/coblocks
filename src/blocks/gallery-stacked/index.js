/**
 * External dependencies
 */
import { GalleryStackedIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import { hasFormattingCategory } from '../../utils/block-helpers';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Block constants
 */
const { name, category } = metadata;

const settings = {
	attributes: metadata.attributes,
	category: hasFormattingCategory ? 'coblocks-galleries' : 'media',
	deprecated,
	/* translators: block description */
	description: __( 'Display multiple images in a single column stacked gallery.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			fullwidth: false,
			images: [
				{ index: 0, url: 'https://s.w.org/images/core/5.3/Windbuchencom.jpg' },
				{ index: 1, url: 'https://s.w.org/images/core/5.3/Glacial_lakes,_Bhutan.jpg' },
			],
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'gallery', 'coblocks' ),
		/* translators: block keyword */
		__( 'photos', 'coblocks' ),
		/* translators: block keyword */
		__( 'lightbox', 'coblocks' ),
	],
	save,
	/* translators: block name */
	title: __( 'Stacked', 'coblocks' ),
	transforms,
};

export { name, category, icon, metadata, settings };
