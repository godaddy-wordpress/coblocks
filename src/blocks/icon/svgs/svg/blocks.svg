<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>