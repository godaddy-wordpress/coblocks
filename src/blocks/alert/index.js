/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
