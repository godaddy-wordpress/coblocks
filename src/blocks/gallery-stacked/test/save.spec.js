/**
 * External dependencies
 */
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

const baseAttributes = {
	images: [
		{ id: 1, url: 'https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg' },
		{ id: 2, url: 'https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg' },
		{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
	],
};

describe( `Tests for ${ name }`, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name, baseAttributes );
		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with images', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( baseAttributes.images[ 0 ].url );
		expect( serializedBlock ).toContain( `data-id="${ baseAttributes.images[ 0 ].id }"` );
		expect( serializedBlock ).toContain( `wp-image-${ baseAttributes.images[ 0 ].id }` );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have className \'has-lightbox\' with lightbox enabled.', () => {
		block.attributes = { ...block.attributes, lightbox: true };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-lightbox' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have className \'has-fullwidth-images\' with Fullwidth Images enabled.', () => {
		block.attributes = { ...block.attributes, fullwidth: true };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-fullwidth-images' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have figcaption element with captions enabled and caption text.', () => {
		block.attributes = {
			...block.attributes,
			captions: false,
			images: [
				...block.attributes.images,
				{ caption: 'test caption', id: 3, url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
			],
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).not.toContain( '<figcaption class="coblocks-gallery--caption">' );

		block.attributes.captions = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '<figcaption class="coblocks-gallery--caption">' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	[ 'none', 'sml', 'med', 'lrg' ].forEach( ( shadow ) => {
		it( `should have className \'has-shadow-${ shadow }\' with box shadow set.`, () => {
			block.attributes = { ...block.attributes, shadow };

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-shadow-${ shadow }` );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should have custom link when linkTo is set to "custom"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ id: 3, imgLink: 'http://google.com', url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
			],
			linkTo: 'custom',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="http://google.com"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have media link when linkTo is set to "media"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ id: 3, url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
			],
			linkTo: 'media',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have attachment link when linkTo is set to "attachment"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ id: 3, link: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
			],
			linkTo: 'attachment',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	[ 'grayscale', 'sepia', 'saturation', 'dim', 'vintage' ].forEach( ( filter ) => {
		it( `should have className \'has-filter-${ filter }\' with filter set to '${ filter }'.`, () => {
			block.attributes = { ...block.attributes, filter };
			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-filter-${ filter }` );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );
} );
