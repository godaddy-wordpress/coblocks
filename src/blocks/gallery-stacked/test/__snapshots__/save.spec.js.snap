// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for coblocks/gallery-stacked should have attachment link when linkTo is set to "attachment" 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"linkTo\\":\\"attachment\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"><img class=\\"wp-image-3 has-shadow-none\\" data-id=\\"3\\" data-link=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-filter-dim' with filter set to 'dim'. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"filter\\":\\"dim\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-filter-dim has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-filter-grayscale' with filter set to 'grayscale'. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"filter\\":\\"grayscale\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-filter-grayscale has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-filter-saturation' with filter set to 'saturation'. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"filter\\":\\"saturation\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-filter-saturation has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-filter-sepia' with filter set to 'sepia'. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"filter\\":\\"sepia\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-filter-sepia has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-filter-vintage' with filter set to 'vintage'. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"filter\\":\\"vintage\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-filter-vintage has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-fullwidth-images' with Fullwidth Images enabled. 1`] = `
"<!-- wp:coblocks/gallery-stacked -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-lightbox' with lightbox enabled. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"lightbox\\":true} -->
<div class=\\"wp-block-coblocks-gallery-stacked has-lightbox\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-shadow-lrg' with box shadow set. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"shadow\\":\\"lrg\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-lrg\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-lrg\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-lrg\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-shadow-med' with box shadow set. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"shadow\\":\\"med\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-med\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-med\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-med\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-shadow-none' with box shadow set. 1`] = `
"<!-- wp:coblocks/gallery-stacked -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have className 'has-shadow-sml' with box shadow set. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"shadow\\":\\"sml\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-sml\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-sml\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-sml\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have custom link when linkTo is set to "custom" 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"linkTo\\":\\"custom\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"http://google.com\\"><img class=\\"wp-image-3 has-shadow-none\\" data-id=\\"3\\" data-imglink=\\"http://google.com\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have figcaption element with captions enabled and caption text. 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"captions\\":true} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-3 has-shadow-none\\" data-id=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/><figcaption class=\\"coblocks-gallery--caption\\">test caption</figcaption></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should have media link when linkTo is set to "media" 1`] = `
"<!-- wp:coblocks/gallery-stacked {\\"linkTo\\":\\"media\\"} -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></a></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></a></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"><img class=\\"wp-image-3 has-shadow-none\\" data-id=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;

exports[`Tests for coblocks/gallery-stacked should render with images 1`] = `
"<!-- wp:coblocks/gallery-stacked -->
<div class=\\"wp-block-coblocks-gallery-stacked\\"><ul class=\\"coblocks-gallery has-no-alignment has-fullwidth-images\\"><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-1 has-shadow-none\\" data-id=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"wp-image-2 has-shadow-none\\" data-id=\\"2\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"coblocks-gallery--item\\"><figure class=\\"coblocks-gallery--figure\\"><img class=\\"has-shadow-none\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-stacked -->"
`;
